### PR TITLE
docs: add authorize decorator

### PR DESCRIPTION
**Pull request by jannyHou**
Signed-off-by: jannyHou <juehou@ca.ibm.com>

connect to https://github.com/strongloop/loopback-next/issues/5142

## Checklist

- [ ] DCO (Developer Certificate of Origin) [signed in all commits](https://loopback.io/doc/en/contrib/code-contrib.html)
- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/sidebars/lb4_sidebar.yml [hunk 1]
```diff
@@ -538,6 +538,10 @@ children:
         url: Decorators_authenticate.html
         output: 'web, pdf'
 
+      - title: 'Authorization Decorator'
```
Comment (raymondfeng) [inline docs/site/sidebars/lb4_sidebar.yml:541]: `Authorization Decorators`?

File: docs/site/decorators/Decorators_authorize.md [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+
+## Authorization Decorator
+
+Syntax:
+
+- `@authorize({resource: 'order', scopes: ['create']})`
+
+The authorization decorator is used to provide access control metadata. As part
+of the component [`@loopback/authorization`](../Authorization-component.md), it
+is applied to controller members and specifies who can perform which operations
```
Comment (dhmlau) [inline docs/site/decorators/Decorators_authorize.md:18]: Suggestion on grammar changes:

```suggestion
is applied to controller members and is used to specify who can perform which operations
```

File: docs/site/decorators/Decorators_authorize.md [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+to the protected resource.
+
+`@authorize` takes in an object in type `AuthorizationMetadata`. The syntax
+example specifies the `resource` and `scopes`. A full list of the available
+configuration properties are:
+
+`allowedRoles`/`deniedRoles`: Define the ACL based roles. It should be an array
```
Comment (dhmlau) [inline docs/site/decorators/Decorators_authorize.md:25]: i think you might mean to add bullets to split up each item?

File: docs/site/decorators/Decorators_authorize.md [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+Syntax:
+
+- `@authorize({resource: 'order', scopes: ['create']})`
+
+The authorization decorator is used to provide access control metadata. As part
+of the component [`@loopback/authorization`](../Authorization-component.md), it
+is applied to controller members and specifies who can perform which operations
+to the protected resource.
+
+`@authorize` takes in an object in type `AuthorizationMetadata`. The syntax
```
Comment (dhmlau) [inline docs/site/decorators/Decorators_authorize.md:21]: Suggested change:
```suggestion
The `@authorize` decorator takes in an object in type `AuthorizationMetadata`. The syntax
```

File: docs/site/decorators/Decorators_authorize.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
...
+    this.orders.push(order);
+    return order.id;
+  }
+}
+```
+
+### Class Level Decorator
+
+To configure a default authorization for all methods within a class,
+`@authorize` can also be applied at the class level. In the code below remote
```
Comment (dhmlau) [inline docs/site/decorators/Decorators_authorize.md:60]:
```suggestion
`@authorize` can also be applied at the class level. In the code below, remote
```

File: docs/site/decorators/Decorators_authorize.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+method `numOfViews()` is protected with `ADMIN` role, while authorization for
+remote method `hello()` is skipped by `@authorize.skip()`.
```
Comment (dhmlau) [inline docs/site/decorators/Decorators_authorize.md:62]:
```suggestion
remote method `hello()` is skipped by the use of `@authorize.skip()`.
```

**Review by dhmlau — APPROVED** [review]
LGTM from the readers' perspective, just had a few minor comments. 
Please get a technical review from other maintainers. Thanks!

File: docs/site/decorators/Decorators_authorize.md (hunk 3 above)
Comment (jannyHou) [reply docs/site/decorators/Decorators_authorize.md:25]: right

**Review by raymondfeng — APPROVED** [review]

**Pull request merged**